frontend/src/services/notebookService.ts

```diff
@@ -290,24 +286,31 @@ export const startNotebook = (data: StartNotebookData): Promise<Notebook> => {
   });
 };
 
-export const stopNotebook = (projectName: string, notebookName: string): Promise<Notebook> => {
+export const stopNotebook = async (
+  projectName: string,
+  notebookName: string,
+): Promise<Notebook> => {
   const dateStr = new Date().toISOString().replace(/\.\d{3}Z/i, 'Z');
-  const patch: RecursivePartial<Notebook> = {
+  const notebook = await getNotebook(projectName, notebookName);
+  const patch: Notebook = _.merge({}, notebook, {
     metadata: { annotations: { 'kubeflow-resource-stopped': dateStr } },
-  };
-
-  return patchNotebook(projectName, notebookName, patch);
+  });
+  return replaceNotebook(projectName, notebookName, patch);
 };
 
-export const patchNotebook = (
+export const replaceNotebook = async (
   projectName: string,
   notebookName: string,
-  updateData: RecursivePartial<Notebook>,
+  data: Notebook,
 ): Promise<Notebook> => {
   const url = `/api/notebooks/${projectName}/${notebookName}`;
+  const notebook = await getNotebook(projectName, notebookName);
+  notebook.spec = data.spec;
+  notebook.metadata.annotations['kubeflow-resource-stopped'] =
+    data.metadata.annotations['kubeflow-resource-stopped'];
 
   return axios
-    .patch(url, updateData)
+    .patch(url, _.merge(notebook, data))
```

Comment:
I do not encourage this kind of mutating code -- this is a bad practice as object references are going to get mutated... but since the flow is from a new object it won't be as problematic -- but as we all know, code gets copied, and this can have undesired consequences if done in the wrong area. 

Also, why not just `_.merge({}, notebook, data)`? Why the extra steps to mutate it first then merge it with the data anyways.

Reply:
So I did this mutation because if I merged the whole notebook object, the settings from the previous notebook (spec) would be carried over (like the gpu settings). If i replaced it completely, important parameters, such as the resource version would not exist. The merge also seems to prefer existing values (so "foo" is not replaced by undefined) so I had to add 2 mutations, replace the spec completely to prevent any settings from carrying over, replace the `kubeflow-resource-stopped` to make sure it is replaced with an empty value and then merge the rest to make sure the API does not complain about the resource version and other metadata missing. Please let me know if there is a cleaner way to approach this. I was thinking about copying over the the metadata and replacing the object whole possibly.